src/runtime/threads/detail/thread_pool_base.cpp

```diff
@@ -47,7 +47,7 @@ namespace hpx { namespace threads { namespace detail
     ///////////////////////////////////////////////////////////////////////////
     mask_cref_type thread_pool_base::get_used_processing_units() const
     {
-        std::lock_guard<pu_mutex_type> l(used_processing_units_mtx_);
+//         std::lock_guard<pu_mutex_type> l(used_processing_units_mtx_);
```

Comment:
That lock is needed for concurrent enabling/disabling of PUs

Reply:
This function is called constantly from a hot loop from inside the scheduler. If the lock is really needed we have to find a way not to use this function as it is accounting for about 50% of the speedup I was seeing (%5 of the overall runtime).

Reply:
Back when I implemented it at first, I measured the same. The real question however is if we support that use case or not. Leaving out the lock might speed up things, but also eliminates a use case.

Reply:
Well the obvious thing to do is have differnet flags set for the schedulers at creation time. If the user knows that they are not hot swapping PUs from pools, then this functionality is not required, we can add it to the background_work and suchlike flags used at scheduler creation time.

Reply:
we should add a 'dynamic_pool' flag (or similar) to the pool or scheduler when it is created and if the user attempts to change it at runtime it can throw if necessary...

Reply:
Alternatively, we could cache the mask inside the scheduler and update it on a 'as needed basis', i.e. only whenever PUs have been enabled/disabled.